app/src/lib/git/status.ts

```diff
@@ -163,12 +163,23 @@ function convertToAppStatus(
       kind: AppFileStatusKind.Copied,
       oldPath,
       submoduleStatus: entry.submoduleStatus,
+      renameIncludesModifications: false,
     }
   } else if (entry.kind === 'renamed' && oldPath != null) {
+    // The renamed files includes modifications if the working tree has
+    // modifications of the rename score is not 100%.
+    let renameIncludesModifications = false;
+    if (
+      entry.workingTree === GitStatusEntry.Modified ||
+      (entry.renameOrCopyScore !== undefined && entry.renameOrCopyScore < 100)
+    ) {
+      renameIncludesModifications = true;
+    }
     return {
       kind: AppFileStatusKind.Renamed,
       oldPath,
       submoduleStatus: entry.submoduleStatus,
+      renameIncludesModifications,
```

Comment:
```suggestion
    return {
      kind: AppFileStatusKind.Renamed,
      oldPath,
      submoduleStatus: entry.submoduleStatus,
      renameIncludesModifications: entry.workingTree === GitStatusEntry.Modified ||
      (entry.renameOrCopyScore !== undefined && entry.renameOrCopyScore < 100),
```

We[ prefer Immutability when possible.](https://github.com/desktop/desktop/blob/development/docs/contributing/engineering-values.md#immutability-is-good)